faunadb-scala/src/main/scala/faunadb/values/Codec.scala

```diff
@@ -145,11 +198,12 @@ object Decoder {
       }
   }
 
-  implicit object ByteArrayDecoder extends Decoder[Array[Byte]] {
+  implicit def EitherDecoder[A, B](implicit a: Decoder[A], b: Decoder[B], tagA: ClassTag[A], tagB: ClassTag[B]): Decoder[Either[A, B]] = new Decoder[Either[A, B]] {
```

Comment:
I think I don't understood what you meant to say with `push any requirements for error handling into the generic decoder`. The only purpose for this is to get the runtime name of the class in order to format a error message `s"Either ${tagA.runtimeClass.getSimpleName} or ${tagB.runtimeClass.getSimpleName}"` is there other way to do it?